pkg/apis/clusterregistry/v1alpha1/types.go

```diff
@@ -23,16 +23,89 @@ import (
 // +genclient
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
 // Information about a cluster in a cluster registry.
 type Cluster struct {
 	metav1.TypeMeta `json:",inline"`
 	// Standard object's metadata.
 	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// State is the state of the cluster. This is not a specification, and is not
+	// meant to be used by actively-reconciling controllers; it is also not
+	// a status, as it contains fields that do not necessarily describe the
+	// status of the cluster, and is not necessarily updated by an active
+	// controller.
+	// +optional
+	State ClusterState `json:"state,omitempty" protobuf:"bytes,1,opt,name=state"`
```

Comment:
This shouldn't be optional

Reply:
Agreed in principle, but see above comment. It looks like many (most? all?) objects mark their Spec and Status as optional, so it seems to make sense to do that here.

Reply:
bizarre and unexpected, but fine for now.